core/shared/src/main/scala/org/virtuslab/yaml/YamlError.scala

```diff
@@ -13,39 +14,49 @@ sealed trait YamlError {
   def msg: String
 }
 
-final case class ParseError(msg: String) extends YamlError
+sealed trait ParseError extends YamlError
 object ParseError {
-  def from(expected: String, got: Token): ParseError = ParseError(
-    s"""|Expected 
+  def from(expected: String, got: Token): ParseError    = ExpectedTokenKind(expected, got)
+  def from(expected: TokenKind, got: Token): ParseError = ParseError.from(expected.toString, got)
+
+  final case class ExpectedTokenKind(expected: String, got: Token) extends ParseError {
+    def msg: String =
+      s"""|Expected 
         |$expected but instead got ${got.kind}
         |${got.range.errorMsg}""".stripMargin
-  )
-  def from(expected: TokenKind, got: Token): ParseError = ParseError.from(expected.toString, got)
+  }
+
+  final case class NoRegisteredTagDirective(handleKey: String, tokenTag: Token) extends ParseError {
+    def msg: String = s"There is no registered tag directive for handle $handleKey"
+  }
 }
 
 final case class ComposerError(msg: String) extends YamlError
 
 final case class ModifyError(msg: String) extends YamlError
```

Comment:
for future: ModifyError is shouldn't rather be part of `YamlError` hierarchy. cc: @lbialy

Reply:
Could you quickly enumerate what each of these error types is meant to represent and when can they occur?